### DISCUSSION TITLE
`mise activate --no-tools`

### DISCUSSION THREAD

**Discussion by jdx** [jdx/mise#3878] [Ideas]
it's come up a couple of times that people want enter/leave hooks but not mise tools. Notably this would work well in my [personal workflow](https://mise.jdx.dev/how-i-use-mise.html).

That said, I'd like to get some use-cases where people would like this before putting code to repo.

**Comment by JamBalaya56562** [comment]
A use-case, plus a note that this is already achievable — so the flag would be ergonomics rather
than new capability.

`[settings] enable_tools = []` — an explicit empty allowlist — disables every tool while leaving
the rest of activation intact. Measured on **v2026.8.3** against one project config
(`[tools] shfmt`, `[env] MY_PROJECT_VAR`, `[hooks] enter`/`leave`):

```
                      MY_PROJECT_VAR    shfmt on PATH
default               set-by-mise       …/installs/shfmt/3.10.0/shfmt
enable_tools = []     set-by-mise       NO
```

and the hooks still fire:

```console
[cd into project]
>>> enter hook fired
[cd out of project]
>>> leave hook fired
```

The use-case I hit: a machine where the tools come from the OS package manager or from a container
image, and mise is wanted purely for per-directory env and hooks. For that, a global-config setting
is arguably the right granularity anyway — it is a property of the machine, not of the invocation.

Where a flag would beat the setting is scope. Two differences, both measured:

- `enable_tools = []` is not activation-scoped. It also empties `mise ls --current` and turns
  `mise install` into `all tools are installed` — for the same config that otherwise reports
  `shfmt 3.10.0 (missing)` and `shfmt@3.10.0 ⇢ would install`. A `--no-tools` on `activate` would
  presumably only stop PATH from being modified, leaving the management commands alone.
- It cannot be scoped to one shell without editing config or exporting a setting env var, so
  `eval "$(mise activate zsh --no-tools)"` in a scratch terminal has no one-liner equivalent.

If either of those matters, that is the argument for the flag; if not, the setting already covers
the workflow.
